token/services/interop/htlc/wallet.go

```diff
@@ -154,6 +241,74 @@ func Wallet(sp view2.ServiceProvider, wallet *token.OwnerWallet, opts ...token.S
 
 	return &OwnerWallet{
 		wallet:       wallet,
+		vault:        vault.TokenVault(),
 		queryService: vault.TokenVault().QueryEngine(),
+		bufferSize:   100,
+	}
+}
+
```

Comment:
what is a PickFunction

Reply:
I have renamed it to SelectorFunction. It is used to select tokens matching certain conditions